frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -67,5 +70,25 @@ export const formatNumber = (number: number, options?: NumberFormatOptions) => {
   return `${prefix}${formattedNumber}${suffix}`;
 };
 
+// Simple hack to handle small decimal numbers (0-1)
+function handleSmallNumberFormat<T>(value: number, options: T): T {
+  if (Math.abs(value) < 1 && Math.abs(value) >= 0) {
+    options = maybeMerge(options, {
+      compact: true,
+      decimals: 4,
```

Comment:
I think numbers smaller than `0.0001` would start to not make sense for most users, and by hard-coding this to 4 decimals we would solve most of the small numbers problems.
![image](https://user-images.githubusercontent.com/1937582/189693752-93f424b1-6f9d-45ce-868d-e05a1f72cd24.png)